### PR TITLE
V.4.0.0 rc1

### PR DESCRIPTION
**Pull request by tmsMedcom**
@ovi-medcom 
Please review profiling: [input/fsh/MedComCareCommunication.fsh](https://github.com/medcomdk/dk-medcom-carecommunication/compare/master...v.4.0.0-rc1#diff-182c98e70f594a254d9444b61b0df627f7027b80890bcead9d5afb6204a3c2f7)
[input/fsh/MedComCareCommunicationProvenance.fsh](https://github.com/medcomdk/dk-medcom-carecommunication/compare/master...v.4.0.0-rc1#diff-b3322ff4dd8b631eeb3380df1fef737248fda26d2d24aca34dd5be3266ef8824)

And profile description: [...pagecontent/StructureDefinition-medcom-careCommunication-communication-intro.md](https://github.com/medcomdk/dk-medcom-carecommunication/compare/master...v.4.0.0-rc1#diff-e94ea6de615c983d98e8ddc18401891a188a09ce08d9e14d92811a171e9b2e06)
 and  [input/pagecontent/StructureDefinition-medcom-careCommunication-provenance-intro.md](https://github.com/medcomdk/dk-medcom-carecommunication/compare/master...v.4.0.0-rc1#diff-a084216645bdebcf8f4877f41ee6cffc447762db20f5aba07a9587dc372b9f1c) 

And example page: [input/pagecontent/examples.md](https://github.com/medcomdk/dk-medcom-carecommunication/compare/master...v.4.0.0-rc1#diff-6181737221e33af651a30a6e2690a7cb36dbf549a26a98f4a4a80f255719ad25)

All changes can be found here: https://build.fhir.org/ig/medcomdk/dk-medcom-carecommunication/branches/v.4.0.0-rc1/ 

@RikkeVestesen, please review the example page and examples, as discussed. 

Please notice that one examples is missing.

### CODE REVIEW

**Review by RikkeVestesen — CHANGES_REQUESTED** [review]
1) https://build.fhir.org/ig/medcomdk/dk-medcom-carecommunication/branches/v.4.0.0-rc1/StructureDefinition-medcom-careCommunication-communication.html er set igennem og kommentarer er i review.

Eksemplerne er gennemgået, kommentarer er i review.

Jeg har desuden rettet lidt korrektur i index.md

Sig til, hvis du har spørgsmål :)

File: input/pagecontent/examples.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+The initial message sent from Hjerteafdelingen på Herlev og Gentofte hospital to Plejecenter Herlev.
+
+The message includes one message segment and a new communication identifier. Since it is a new message the activity is in the Provenance instance 'new-message'. The sender and receiver organizations are both referenced from the MessageHeader. 
+
+The CareCommunication contains a topic at Communication.topic.
+
+### 2nd message - reply CareCommunication
+
+[2nd message](Bundle-b56549f7-ed10-422d-8088-f7222b686e46.html)
```
[the remark this answers is not in the document]
Comment (tmsMedcom) [reply input/pagecontent/examples.md:27]: Super - Det ses ved, at der ikke er en reference til forfatteren i det payload, der indeholder bilaget.

File: input/pagecontent/examples.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+### 3rd message - reply CareCommunication
+
+[3rd message](Bundle-3dcb5618-3055-406a-9034-1b8fc8de0fea.html) 
+
+This is a reply to the 2nd message. Therefore, the Communication.identifier remains the same. One Communication.payload has been added with the reply. The base64-encoded content from the attachment in the 2nd message is not included, only the identifier and author information.
+The unique ids of the instances are updated, however the referenced MessageHeader in the Provenance instances from the previous messages remain the same. 
+
+### 4th message - forward CareCommunication
+
+[4th message](Bundle-c0426e3e-978f-46e8-a366-a30f27854b0a.html)
```
[the remark this answers is not in the document]
Comment (tmsMedcom) [reply input/pagecontent/examples.md:43]: I MessageHeaderen skal du se efter, hvilken organization, der er afsender. Når du så kigger på afsender organisationen, skal elementet 'name' ikke fremgå. :)

File: input/pagecontent/examples.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+### 4th message - forward CareCommunication
+
+[4th message](Bundle-c0426e3e-978f-46e8-a366-a30f27854b0a.html) 
+
+This is a forwarding of the 3rd message. Therefore is the Communication.identifier updated and the base64-encoded content from the attachment in the 2nd message is included. Additionally, is a Communciation.payload added with a reason for the forwarding. The topic is not forwarded. The activity in the Provenance instance is 'forward-message'. A new receiver organization is included. The sender organization does not include a name in Organization.name.
+
+### 5th message - reply CareCommunication
+
+[5th message](Bundle-d11968f5-4bdf-4b50-b146-a8e1cc890fc3.html)
```
[the remark this answers is not in the document]
Comment (tmsMedcom) [reply input/pagecontent/examples.md:49]: Ja, nemlig. Der er en fejl i beskrivelsen. Det er PractitionerRole.code.text (code tilføjet)

File: input/pagecontent/StructureDefinition-medcom-careCommunication-communication-intro.md [hunk 4]
```diff
@@ -1,47 +1,37 @@
...
-
-When sending a CareCommunication message it is possible to add a more specific receiver of the message, called a recipient, and a more specific sender, which can be found in the elements Communication.recipient and Communication.extension.sender, respectively. It is allowed to reference a [MedComCoreCareTeam](https://medcomfhir.dk/ig/core/StructureDefinition-medcom-core-careteam.html), which is people or an organization participating a coordination or delivery of patientcare, or [MedComCorePractitionerRole](https://medcomfhir.dk/ig/core/StructureDefinition-medcom-core-practitionerrole.html), which references a MedComCorePractitioner to address a healthcare professional involved in patient care. Common for both MedComCoreCareTeam and MedComCorePractitionerRole/MedComCorePractitioner is that a name of the careteam or practitioner should be included as well as a reference to a MedComCoreOrganization which represents the sender or receiver defined in [MedComCareCommunicationMessageHeader](https://medcomfhir.dk/ig/carecommunication/StructureDefinition-medcom-careCommunication-messageHeader.html).
```
[the remark this answers is not in the document]
Comment (tmsMedcom) [reply input/pagecontent/StructureDefinition-medcom-careCommunication-communication-intro.md:32]: Jeg har flyttet rundt på indholdet på siden, men ser lidt voldsomt ud :)

File: input/pagecontent/StructureDefinition-medcom-careCommunication-communication-intro.md [hunk 5]
```diff
@@ -92,14 +82,28 @@ R = required and O = optional.
...
 **Relevant phone number:** A relevant phone number e.g. to the department from which the CareCommunication is sent. The phone number should be applied automatically. In case it is not possible to apply the phone number automatically, it shall be applied by the author. <br> 
 **Timestamp:** Represents the real world event, where the user presses "send" to send the CareCommunication.<br> 
-**Identifier:** An UUID version 4 with a reference to the assigner organisation. The assigner organisation is the organisation who assigned the UUID, which will be the same as the initial sender organisation of the message segment. If the attachment is modified and reattached, it shall be given a new UUID and the assigner shall be updated. <br> 
+**Identifier:** An UUID version 4 with a reference to the assigner organisation. The assigner organisation is the organisation who assigned the UUID, which will be the same as the sender organisation of the message segment. <br>
```
[the remark this answers is not in the document]
Comment (tmsMedcom) [reply input/pagecontent/StructureDefinition-medcom-careCommunication-communication-intro.md:88]: God pointe!

File: input/pagecontent/index.md
[the remark this answers is not in the document]
Comment (tmsMedcom) [reply input/pagecontent/index.md:1]: Tak for hjælpen @RikkeVestesen - rigtig gode bemærkninger! :D

**Review by ovi-medcom — APPROVED** [review]

**Comment by tmsMedcom** [comment]
@ovi-medcom thank you! Have you changed the bug or what is it?

**Comment by ovi-medcom** [comment]
In the MedComMessaging IG I would have expected MedComMessagingMessageHeader  to require a MedComMessagingMessageDefinitionfhir.

When referencing a [MedComMessagingMessageDefinition] canonical "version" must be included, like "[MedComMessagingMessageDefinition.canonicalurl] | [MessageDefinition.version]]" where content in brackets are replaced by the concrete content of these elements and the two elements are  separated by the "|".

In MedComCareCommunicationMessageHeader MedComCareCommunicationMessageDefinition should be referenced instead of MessageDefinition and with the proper version 

Hope this is clear enough :-)

**Comment by ovi-medcom** [comment]
Another thing we need to move to MedComMessaging IG is this valueset: https://build.fhir.org/ig/medcomdk/dk-ehmi-eer/ValueSet-ehmi-messagedefinition-use-context-code-valueset.html

The name of the valueset should be changed to something like medcom-messagedefinition-use-context-code-valueset. 
After this the reference to the eer-valueset in https://build.fhir.org/ig/medcomdk/dk-medcom-messaging/StructureDefinition-MedComMessagingMessageDefinition.html should be changed

**Comment by tmsMedcom** [comment]
@ovi-medcom

> In the MedComMessaging IG I would have expected MedComMessagingMessageHeader to require a MedComMessagingMessageDefinitionfhir.
> 
I haven't made changes to the MedComMessagingIG. I guess I assumed it was good to go. But I will make the changes.

> When referencing a [MedComMessagingMessageDefinition] canonical "version" must be included, like "[MedComMessagingMessageDefinition.canonicalurl] | [MessageDefinition.version]]" where content in brackets are replaced by the concrete content of these elements and the two elements are separated by the "|".
> 
Is this what you expect? 
> In MedComCareCommunicationMessageHeader MedComCareCommunicationMessageDefinition should be referenced instead of MessageDefinition and with the proper version
> 

Is this what you expect? 
In profile: image

and in instance: 
image

@

**Comment by tmsMedcom** [comment]
@ovi-medcom 

> Another thing we need to move to MedComMessaging IG is this valueset: https://build.fhir.org/ig/medcomdk/dk-ehmi-eer/ValueSet-ehmi-messagedefinition-use-context-code-valueset.html
> 
> The name of the valueset should be changed to something like medcom-messagedefinition-use-context-code-valueset. After this the reference to the eer-valueset in https://build.fhir.org/ig/medcomdk/dk-medcom-messaging/StructureDefinition-MedComMessagingMessageDefinition.html should be changed

Shouldn't the terminology be kept in the terminology IG?

**Comment by ovi-medcom** [comment]
You have nailed it in both answers, the responsibility of not being clear on, what you could assume, is mine :-)
Thanks!
Ready to go?

**Comment by tmsMedcom** [comment]
Perfect! Yes - ready to go

**Pull request merged**